File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    for i in range(len(babbling)):
        for cont in ['ayaaya','yeye','woowoo','mama']:
            if cont in babbling[i]:
                babbling[i]='x'
        for coo in ['aya','ye','woo','ma']:
            babbling[i]=babbling[i].replace(coo,' ')    
        babbling[i]=babbling[i].replace(' ','')
    return babbling.count('')

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<string> babbling) {
    int r=0;
    vector<string> pronoun={"aya", "ye", "woo", "ma"};
    vector<string> duplePronoun={"ayaaya","yeye","woowoo","mama"};
    for(int i=0;i<babbling.size();i++) {
        for(int j=0;j<duplePronoun.size();j++) {
            if(babbling[i].find(duplePronoun[j])==string::npos) {
                while(babbling[i].find(pronoun[j])!=string::npos) {
                    babbling[i].insert(babbling[i].find(pronoun[j])," ");
                    babbling[i].erase(babbling[i].find(pronoun[j]),pronoun[j].length());
                }
            } else {
                babbling[i]="x";
                break;
            }
        }
        while(babbling[i].find(" ")!=string::npos) {
            babbling[i].erase(babbling[i].find(" "),1);
        }
    }
    for(int i=0;i<babbling.size();i++) {
        if(babbling[i]=="")
            r+=1;
    }
    return r;
}